src/plot_api/subroutines.js

```diff
@@ -324,6 +329,48 @@ exports.lsInner = function(gd) {
     return gd._promises.length && Promise.all(gd._promises);
 };
 
+function findMainSubplot(ax, fullLayout) {
```

Comment:
@alexcjohnson I didn't bother the "find main subplot" algo itself as you proposed in https://github.com/plotly/plotly.js/issues/2549#issuecomment-382775857, I was happy enough with skipping this whole thing except when sploms with `showlowerhalf: false` with https://github.com/plotly/plotly.js/pull/2628/commits/c947b2c0070833199ceff56fc4bafc59d539f3d1